command/agent/local.go

```diff
@@ -379,6 +379,12 @@ func (l *localState) setSyncState() error {
 		}
 
 		// If our definition is different, we need to update it
+		if existing.EnableTagDrift {
+			l.logger.Printf("[DEBUG] Tag drift enabled.")
+			existing.Tags = service.Tags
```

Comment:
This does seem a bit odd though. I think it might be better to have an equality checker function rather than leaning on `reflect.DeepEqual` here. The current logic will also become problematic when switching to an in-memory state store (which is on our short-term roadmap), since the service is a pointer and you are actually swapping out part of the data here. Having an equality check function would also make this easier to test.